recipes/libnet/all/conanfile.py

```diff
@@ -0,0 +1,84 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class LibnetConan(ConanFile):
+    name = "libnet"
+
+    description = "Libnet is an API to help with the construction and injection of network packets."
+    topics = ("conan", "libnet", "network")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://libnet.sourceforge.net/"
+    license = ["BSD-2-Clause"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+    
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("libnet is not supported by Visual Studio")
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("libnet can't be built as shared on Windows")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "libnet-{0}".format(self.version)
+        os.rename(extracted_folder, self._source_subfolder)
```

Comment:
this is our improved version

```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```